packages/miniflare/src/workers/core/proxy.worker.ts

```diff
@@ -132,6 +133,14 @@ export class ProxyServer implements DurableObject {
 	}
 
 	async #fetch(request: Request) {
+		// Validate `Host` header
+		const hostHeader = request.headers.get("Host");
+		if (hostHeader == null) return new Response(null, { status: 400 });
+		const hostname = hostHeader.substring(0, hostHeader.lastIndexOf(":"));
```

Comment:
Is the reason we are not actually parsing the URL here to extract the hostname for performance?